test/UnitTest/Microsoft.AspNetCore.OData.Test/Extensions/ODataApplicationBuilderExtensionsTest.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.OData.Edm;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.OData.Test.Extensions
+{
+    public class ODataApplicationBuilderExtensionsTest
+    {
+        [Fact]
+        public void UseOData_ThrowsInvalidOperationException_IfODataServiceIsNotRegistered()
+        {
+            // Arrange
+            var applicationBuilderMock = new Mock<IApplicationBuilder>();
+            applicationBuilderMock.Setup(s => s.ApplicationServices).Returns(Mock.Of<IServiceProvider>());
+            var edmModel = new Mock<IEdmModel>().Object;
+
+            // Act & Assert
+            var exception = Assert.Throws<InvalidOperationException>(() => applicationBuilderMock.Object.UseOData(edmModel));
+
+            Assert.Equal("Cannot find 'OData services'. The OData services have not been configured. Are you missing a call to AddOData()?",
+                exception.Message);
```

Comment:
Can we use SRResources.MissingODataServices here?

Reply:
Changed.